tls/s2n_handshake_io.c

```diff
@@ -223,7 +286,7 @@ static int handshake_read_io(struct s2n_connection *conn)
     }
 
     if (isSSLv2) {
-        if (conn->handshake.state != CLIENT_HELLO) {
+        if (conn->handshake.message_number != 0) {
```

Comment:
Should we switch this to `ACTIVE_MESSAGE(conn) != CLIENT_HELLO` for consistency?